docs/general/how-ra-works.md

```diff
@@ -0,0 +1,45 @@
+# How RetroAchievements Works
+
+RetroAchievements (RA) provides users the ability to earn achievements in retro games from an RA achievement set.  It awards achievements by comparing a game’s memory, henceforth referred to as RAM, to achievement code written by an RA developer.  Achievement code, also known as logic, is a list of memory conditions chosen by the developer, that when all are simultaneously true on a single frame, will award an achievement.
+
+## What Is an Achievement Set?
+
+An achievement set is the compilation of individual achievements, Rich Presence and leaderboards for a particular game.  Achievements are comprised of achievement logic, a title, a description, a point value and a badge.  Additionally, a set must contain a Rich Presence script which provides site users information on where active players are in a game.  Sets may have leaderboards that track how well players perform in certain things in a game such as how quickly they can beat stages, how many points they can score and many other things.  RA users interface with all these components and subcomponents.  Each one is a vital part of the player experience.
```

Comment:
Some of this may be worth revising for the sake of longevity. Here's an excerpt from the multiset design doc concerning rich presence:

<img width="654" alt="Screenshot 2024-06-06 at 8 02 35 AM" src="https://github.com/RetroAchievements/docs/assets/3984985/4c22bf44-1851-4648-aa19-f01585779ca7">

For practical purposes, how I think this will ultimately evolve is RP scripts will indeed be associated to core and specialty sets, but they will not be associated to bonus sets.

In other words, there is a future state where some sets won't be eligible to have their own RP scripts (bonus sets).

Reply:
Great catch.  Removed rich presence from the list of set components.  Changed "a set must contain a Rich Presence script" to "a core set should contain a Rich Presence script"